<?xml version='1.0' encoding='utf-8' standalone='no'?>
<!DOCTYPE issue SYSTEM "lwg-issue.dtd">

<issue num="3336" status="New">
<title>How does <tt>std::vformat</tt> handle exception thrown by formatters?</title>
<section><sref ref="[format.functions]"/></section>
<submitter>Tam S. B.</submitter>
<date>11 Nov 2019</date>
<priority>2</priority>

<discussion>
<p>
The specification for <tt>std::vformat</tt> in <sref ref="[format.functions]"/>/7 says
</p>
<blockquote>
<p>
<i>Throws:</i> <tt>format_error</tt> if <tt>fmt</tt> is not a format string.
</p>
</blockquote>
<p>
It seems unclear whether <tt>vformat</tt> throws when an exception is thrown by the <tt>formatter</tt>, 
e.g. when the format string is valid, but the corresponding argument cannot be formatted with the given 
format string.
<p/>
For example, the <tt>"c"</tt> format specifier is specified to throw <tt>format_error</tt> if the 
corresponding argument is not in the range of representable values for <tt>charT</tt> 
(Table 60 [tab:format.type.int]). It seems unclear whether <tt>vformat</tt> propagates this exception.
<p/>
It also appears unclear whether <tt>vformat</tt> may throw other types of exception (e.g. <tt>bad_alloc</tt>) 
when the formatter or the constructor of <tt>std::string</tt> result throws.
</p>

<note>2019-11-20 Issue Prioritization</note>
<p>Priority to 2 after reflector discussion.</p>

</discussion>

<resolution>
</resolution>
</issue>
